be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -207,17 +62,23 @@ class SinkBuffer {
     // use int64_t as key, which is the field type of TUniqueId::lo
     // because TUniqueId::hi is exactly the same in one query
 
-    phmap::flat_hash_map<int64_t, size_t> _num_sinkers;
-    phmap::flat_hash_map<int64_t, int64_t> _request_seqs;
-    std::atomic<int32_t> _num_in_flight_rpc = 0;
+    phmap::flat_hash_map<int64_t, int64_t> _num_sinkers;
+    phmap::flat_hash_map<int64_t, int64_t> _request_sequences;
+    // distribution of received response sequence numbers:
+    // part1: { seq | 1 <= seq <= _max_processed_sequence }
+    // part2: { seq | seq = _max_processed_sequence + i, i > 1 }
+    phmap::flat_hash_map<int64_t, int64_t> _max_processed_sequences;
+    phmap::flat_hash_map<int64_t, std::unordered_set<int64_t>> _unprocessed_sequences;
```

Comment:
Why need to keep _unprocessed_sequences?

Reply:
Considering the following situation
1. sending packets `1, 2, 3` in order
2. one possible order of response is `1, 3, 2`, when receive `response-1`, we can update `_max_processed_sequence` to 1, but when receive `response-3`, we cannot update `_max_processed_sequence`. Then receive the final `response-2`, we can update `_max_processed_sequence` from `1` to `3` in one time according to side information stored in `_unprocessed_sequences`

Reply:
Could we use vector, not std::unordered_set here?

Reply:
We need to find specified sequence, if use vector, we could achieve this by maintaining order when inserting or traversing when searching, which seems a little complicated, so I replace it with set